salt/modules/nagios_json.py

```diff
@@ -0,0 +1,189 @@
+# -*- coding: utf-8 -*-
+'''
+Check Host & Service status from Nagios via JSON.
+'''
+
+# Import python libs
+from __future__ import absolute_import
+import logging
+import requests
+
+# Import 3rd-party libs
+from requests.exceptions import ConnectionError
+# pylint: disable=import-error,no-name-in-module,redefined-builtin
+from salt.ext.six.moves.urllib.parse import urljoin as _urljoin
+# pylint: enable=import-error,no-name-in-module
+
+log = logging.getLogger(__name__)
+
+
+def __virtual__():
+    '''
+    Only load if nagios-plugins are installed
+    '''
+    return 'nagios_json'
```

Comment:
Makes sense.  Updated the pull request.  I'll update some of my recent additions taking this into account.